subworkflows/local/pair_variant_calling.nf

```diff
@@ -72,172 +49,76 @@ workflow PAIR_VARIANT_CALLING {
-            BGZIP_VC_MANTA_DIPLOID(MANTA_SOMATIC.out.diploid_sv_vcf)
-            BGZIP_VC_MANTA_SOMATIC(MANTA_SOMATIC.out.somatic_sv_vcf)
-
-            manta_sv_vcf_to_concat = BGZIP_VC_MANTA_SV.out.vcf.map{ meta, vcf ->
-                new_meta = meta.clone()
-                new_meta.id = new_meta.tumor_id + "_vs_" + new_meta.normal_id
-                [new_meta, vcf]
-            }.groupTuple(size: num_intervals)
-
-            manta_small_indels_vcf_to_concat = BGZIP_VC_MANTA_SMALL_INDELS.out.vcf.map{ meta, vcf ->
-                new_meta = meta.clone()
-                new_meta.id = new_meta.tumor_id + "_vs_" + new_meta.normal_id
-                [new_meta, vcf]
-            }.groupTuple(size: num_intervals)
-
-            manta_diploid_vcf_to_concat = BGZIP_VC_MANTA_DIPLOID.out.vcf.map{ meta, vcf ->
-                new_meta = meta.clone()
-                new_meta.id = new_meta.tumor_id + "_vs_" + new_meta.normal_id
-                [new_meta, vcf]
-            }.groupTuple(size: num_intervals)
-
-            manta_somatic_sv_vcf_to_concat = BGZIP_VC_MANTA_SOMATIC.out.vcf.map{ meta, vcf ->
-                new_meta = meta.clone()
-                new_meta.id = new_meta.tumor_id + "_vs_" + new_meta.normal_id
-                [new_meta, vcf]
-            }.groupTuple(size: num_intervals)
-
-            CONCAT_MANTA_SV(manta_sv_vcf_to_concat, fasta_fai, intervals_bed_combine_gz)
-            CONCAT_MANTA_SMALL_INDELS(manta_small_indels_vcf_to_concat,fasta_fai, intervals_bed_combine_gz)
-            CONCAT_MANTA_DIPLOID(manta_diploid_vcf_to_concat, fasta_fai, intervals_bed_combine_gz)
-            CONCAT_MANTA_SOMATIC(manta_somatic_sv_vcf_to_concat, fasta_fai, intervals_bed_combine_gz)
-
-            manta_candidate_small_indels_vcf = CONCAT_MANTA_SV.out.vcf
-            manta_candidate_sv_vcf           = CONCAT_MANTA_SMALL_INDELS.out.vcf
-            manta_diploid_sv_vcf             = CONCAT_MANTA_DIPLOID.out.vcf
-            manta_somatic_sv_vcf             = CONCAT_MANTA_SOMATIC.out.vcf
-
-            ch_versions = ch_versions.mix(BGZIP_VC_MANTA_SV.out.versions)
-            ch_versions = ch_versions.mix(BGZIP_VC_MANTA_SMALL_INDELS.out.versions)
-            ch_versions = ch_versions.mix(BGZIP_VC_MANTA_DIPLOID.out.versions)
-            ch_versions = ch_versions.mix(BGZIP_VC_MANTA_SOMATIC.out.versions)
-
-            ch_versions = ch_versions.mix(CONCAT_MANTA_SV.out.versions)
-            ch_versions = ch_versions.mix(CONCAT_MANTA_SMALL_INDELS.out.versions)
-            ch_versions = ch_versions.mix(CONCAT_MANTA_DIPLOID.out.versions)
-            ch_versions = ch_versions.mix(CONCAT_MANTA_SOMATIC.out.versions)
-
+    cram_pair_intervals = cram_pair.combine(intervals)
+        .map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai, intervals ->
+            normal_id = meta.normal_id
+            tumor_id = meta.tumor_id
+            new_intervals = intervals.baseName != "no_intervals" ? intervals : []
```

Comment:
what about using `num_intervals` as we do in the preprocessing?

Reply:
Right.... maybe we can then actually get rid completely of the no_intervals file crutch file. This was its only purpose as far as I can tell, I think because for some time we needed an empty file as input, because emtpy list wasn't an option as we do now all over the place

Reply:
:exploding_head:

Reply:
Maybe we can just fix the intervals thingy in a separate PR